internal/transport/transport_test.go

```diff
@@ -2191,15 +2191,10 @@ func runPingPongTest(t *testing.T, msgSize int) {
 	binary.BigEndian.PutUint32(outgoingHeader[1:], uint32(msgSize))
 	opts := &Options{}
 	incomingHeader := make([]byte, 5)
-	done := make(chan struct{})
-	go func() {
-		timer := time.NewTimer(time.Second * 5)
-		<-timer.C
-		close(done)
-	}()
+	timer := time.NewTimer(time.Second * 1)
 	for {
 		select {
-		case <-done:
+		case <-timer.C:
```

Comment:
This loop is a little awkward.

Instead, how about:

```go
ctx, cancel = context.WithTimeout(ctx, time.Second)
defer cancel()
for ctx.Err() == nil {
	// do the stuff inside "default"
}
// do the stuff inside <-timer.C
```

(Or don't reuse the original `ctx` if that's too icky to overwrite it. Or use a channel closed by an `AfterFunc`, or use a `grpcsync.Event` that's fired by one.)

Reply:
I'm not a huge fan of that `select` inside the infinite `for` loop...it makes it harder to understand than the structure I mentioned before.  But it's fine to leave it as-is in this kind of PR.

Reply:
Actually, can you just change it back to your timer?  That was cleaner than the current way.  The point of my suggestion was to remove the `select` from the loop entirely.

Reply:
I updated the PR to use the override the ctx timeout to 1s. This seems more cleaner now. first this happens, and when after the timeout this.